File: JH/DAY11/10610.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    string s;
    cin >> s;
    std::sort(s.begin(), s.end(),greater<>());


    if(s[s.length()-1] != '0'){
        cout << -1;
    }else{
        int sum=0;
        for (int i = 0; i <s.length()-1 ; ++i) {
            sum += s[i]-'0';
            //cout << sum <<"\n";

        }
        if(sum%3==0){
            cout << s;
        }else{
            cout << -1;
        }
    }


}

File: HY/Day 7/10610.py
n = list(input())
#리스트를 내림차순하고 정수형으로 바꿈
n.sort(reverse=True)
n = list(map(int, n))
#30의 배수조건: 3의 배수(모든 자리 수의 합이 3의 배수) and  10의 배수여야함 (가장 작은 수가 0이어야함)
if sum(n) % 3 == 0 and  n[-1] == 0 :
  ##리스트를 다시 하나로 합치고 string으로 변환
  print(''.join(map(str, n )))
  #30의 배수가 아닐경우 -1
else :
  print(-1)

